swagger.yaml

```diff
@@ -6,373 +6,1408 @@ info:
-      content:
-        application/json:
-          schema:
-            $ref: '#/components/schemas/Error'
+      Success:
+        description: Success
+      UserCreated:
+        description: New User Created
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      LoginSuccess:
+        description: Logged In Successfully
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      OwnUserRetrieved:
+        description: User sees own profile
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/User'
+      ErrorBadRequest:
+        description: Bad Request
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
+      ErrorUnauthorized:
+        description: Unauthorized
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
+      ErrorForbidden:
+        description: Forbidden
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
+      ErrorNotFound:
+        description: Not Found
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
+      ErrorUnsupportedMediaType:
+        description: Unsupported Media Type
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
+      ErrorInternalServer:
+        description: Internal Server Error
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Error'
   securitySchemes:
     token:
       type: http
       scheme: bearer
       bearerFormat: JWT
 paths:
-  /users/login/:
+  /auth/signup:
     post:
       tags:
         - Authentication
-      summary: Perform password-based authentication
-      description: Returns a JWT upon success
+      summary: Sends token to email for confirmation
+      description: Returns a JWT upon successful signup. Sends token to email
       requestBody:
         description: User Credentials
         required: true
         content:
           application/json:
             schema:
               properties:
+                fullName:
+                  type: string
+                userName:
+                  type: string
                 email:
                   type: string
                   format: email
                 password:
                   type: string
+                  format: password
+                confirmPassword:
+                  type: string
+                  format: password
       responses:
-        '200':
+        '201':
           $ref: '#/components/responses/LoginSuccess'
+        '409':
+          $ref: '#/components/responses/ErrorBadRequest'
```

Comment:
Nice work, Shola. I think the status code should 400 for a bad request. Kindly fix this.